src/site/content/en/blog/webcodecs/index.md

```diff
@@ -50,73 +50,79 @@ streaming, etc.
+- `MediaStreamTrackProcessor` breaks up into media tracks into individual frames
+- `MediaStreamTrackGenerator` helps to create a media track from a stream of frames.
+
+
+## WebCodecs and web workers
 
-It all starts with a `VideoFrame`. There are two ways to convert existing
-pictures into `VideoFrame` objects.
+By design WebCodecs API does all the heavy lifting asynchronously and off the main thread,
+but since frame and chunk callbacks can often be called multiple times a second,
+they might clutter the main thread and thus make the website less responsive.
+Therefore it is preferable to move handling of individual frames and encoded chunks into a
+web worker.
 
-The first is to create a frame directly from an
-[`ImageBitmap`](https://developer.mozilla.org/en-US/docs/Web/API/ImageBitmap).
-Just call the `VideoFrame()` constructor and give it a bitmap and a presentation
-timestamp.
+To help with that, [ReadableStream](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream)
+provides a convenient way to automatically transfer all frames coming from a media
+track to the worker. For example, `MediaStreamTrackProcessor` can be used to obtain a
+`ReadableStream` for a media stream track coming from the web camera, after that
+the stream is transferred to a web worker where frames are read one by one and queued
+into a `VideoEncoder`.
+
+With [`HTMLCanvasElement.transferControlToOffscreen`](https://developers.google.com/web/updates/2018/08/offscreen-canvas#unblock_main_thread) even rendering can be done off the main thread. But if all the hight level tools turned
+out to be inconvenient, `VideoFrame` itself is [transferable](https://developer.mozilla.org/en-US/docs/Web/API/Transferable) and may be
+moved between workers.
+
+## WebCodecs in action
+
+### Encoding
 
+It all starts with a `VideoFrame`.
+There are three ways to construct video frames
+1. From an image source like canvas, image bitmap or a video element.
 ```js
-let cnv = document.createElement('canvas');
+const cnv = document.createElement('canvas');
 // draw something on the canvas
 …
-let bitmap = await createImageBitmap(cnv);
-let frame_from_bitmap = new VideoFrame(bitmap, { timestamp: 0 });
+let frame_from_canvas = new VideoFrame(cnv, { timestamp: 0 });
 ```
+2. Use `MediaStreamTrackProcessor` to pull frames from a [`MediaStreamTrack`](https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamTrack)
+```js
+  const stream = await navigator.mediaDevices.getUserMedia({ … });
+  const track = stream.getTracks()[0];
+
+  const media_processor = new MediaStreamTrackProcessor(track);
+
+  const reader = media_processor.readable.getReader();
+  while (true) {
+      const result = await reader.read();
+      if (result.done)
+        break;
+      let frame_from_camera = result.value;
+  }
+```
+3. Create a frame from its binary pixel representation in a [`BufferSource`](https://developer.mozilla.org/en-US/docs/Web/API/BufferSource)
```

Comment:
Probably needs a body/example, right now it looks like the figures below are the explanation.